src/AutoRest.CSharp/Common/Utilities/StringExtensions.cs

```diff
@@ -346,7 +346,7 @@ public static string ToSingular(this string plural, bool inputIsKnownToBePlural
         public static string LastWordToSingular(this string plural, bool inputIsKnownToBePlural = true)
         {
             var words = plural.SplitByCamelCase();
-            var lastWord = words.LastOrDefault();
+            var lastWord = words.Last();
```

Comment:
I think we want to leave LastOrDefault here, we never dereference `lastWord` and the comparison below should work with a null string fine.